src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -164,10 +164,12 @@ export class Kernel implements IKernel {
         await promise;
         return promise;
     }
-    public async executeHidden(code: string): Promise<nbformat.IOutput[]> {
+    public async executeHidden(code: string, doc: NotebookDocument): Promise<nbformat.IOutput[]> {
         const stopWatch = new StopWatch();
         const notebookPromise = this.startNotebook();
-        const promise = notebookPromise.then((nb) => executeSilently(nb.session, code));
+
+        const promise = this.kernelExecution.executeHidden(notebookPromise, code, doc);
```

Comment:
should we do this for all the uses of `executeSilently`?

Reply:
No we should not, this has the potential to break variable viewer and others.
I think we'd should have two methods.
And we might want to rename the executeCell to queueAndExecuteCell, 
Similarly executeHidden and queueAndExeceuteHidden

Reply:
We'd should not because if we have 10 cells and we're thinking all, then variables will never refresh because those hidden requests will get queued and run only after all 10 cells finish. Similarly plenty of other pitfalls.
Never this new method should be banned to make it obvious how the code will be executed, ie the fact that it's queued.